network/discovery/dv5_service.go

```diff
@@ -56,6 +57,14 @@ type DiscV5Service struct {
 	dv5Listener Listener
 	bootnodes   []*enode.Node
 
+	// withExtraFiltering specifies whether we want to additionally filter discovered peers,
+	// since operator node and boot node are using the same discovery code (`DiscV5Service`) we
+	// want to do this extra filtering for operator node only
+	withExtraFiltering bool
+	// topicsCtrl must be non-nil when withExtraFiltering is set to true since it will be used
+	// to help us decide which peers need to be filtered out
+	topicsCtrl topics.Controller
```

Comment:
I haven't looked into how boot-node uses discovery, but we are forced to handle this somehow (which `withExtraFiltering` flag is for).